prow/cmd/checkconfig/main_test.go

```diff
@@ -527,12 +537,18 @@ size:
 			got := validateUnknownFields(tc.cfg, tc.configBytes, tc.filename)
 			if !reflect.DeepEqual(got, tc.expectedErr) {
```

Comment:
This is fragile to changes in the YAML library's error formatting.
Instead of trying to match the errors with `DeepEqual` (which requires identical types and error message content), consider checking that the error message text contains some key words that are more likely to remain if the YAML library changes it's error formatting. e.g. match if the error message contains "unknown field" and the unknown field name itself. It may also be appropriate to simply check that an error is returned and use happy path test cases to increase confidence that we aren't getting errors for some other reason.

Reply:
Thanks for the feedback Cole. I'll implement that.
/hold

Reply:
@cjwagner I took a stab at this but only succeeded in making the test uglier and more fragile when matching the three expected substrings and also catering for the `nil` case. I guess the yaml library error is pretty stable, but if not the change would be contained entirely in https://github.com/kubernetes/test-infra/pull/15961/files#diff-1bccbd8d60a651295f7c8bb39ac87d4dR548 and 549.
/unhold

Reply:
Since this prints the error diff if they aren't equal it shouldn't be too burdensome to fix this test when upgrading the YAML library dependency in the rare case that this changes. That being said, I don't really see why using an equality check function that checks for substrings and handles the nil case would be uglier or more fragile than using the equality operator. If you think you can do that cleanly, that would be preferable since it makes fewer assumptions about dependency implementation, but if you just want to merge this use `/hold cancel`.

Reply:
you're totally right, this was a good exercise for me. Pushed in 7fc20d3, would appreciate another set of eyes/lgtm and then you could unhold